src/player.cpp

```diff
@@ -13086,7 +13086,7 @@ void player::add_known_trap(int x, int y, const std::string &t)
 
 bool player::is_deaf() const
 {
-    return has_effect("deaf") || worn_with_flag("DEAF");
+    return has_effect("deaf") || worn_with_flag("DEAF") || (has_active_bionic("bio_earplugs") && !has_active_bionic("bio_ears"));
```

Comment:
This may need a newline as it runs long.  Probably best to put it after the AND.